scripts/get_famous_people_photos/src/get_famous_people_photos.py

```
@@ -0,0 +1,277 @@
+                is_match, distance = identifier.compare_embedding(
+                    anchor_embedding, other.embedding)
+                if is_match:
+                    urls.add(other.url)
+            identifier.tear_down()
+            del identifier
+        except Exception as e:
+            print(e)
+    # Update counter
+    increment()
+    timer.update(int(counter.value))
+    return list(urls), person
+
+
+def dedupe_images(matched_urls, person):
+    """Hashes images and removes nearly identical images
+
+    Arguments:
+        matched_urls {str[] or set} -- list or set of urls to dedupe
+        person{str} -- name of person
+
+    Returns:
+        str[] -- list of deduped urls
+    """
+    image_hashes = [IMAGE_HASH(url_to_img_hash(url), url)
+                    for url in matched_urls]
+    tree = pybktree.BKTree(image_distance, image_hashes)
+    # this makes images saved in order of similarity so we can spot duplicates easier
+    sorted_image_hashes = sorted(tree)
+    to_discard = []
+    urls_to_keep = set()
+    for image_hash in sorted_image_hashes:
+        if image_hash not in to_discard:
+            # gets pictures within a hamming distance of 3
+            matches = tree.find(image_hash, 3)
+            for match in matches:
+                if match[1].url != image_hash.url:
+                    to_discard.append(match[1])
+            urls_to_keep.add(image_hash.url)
+    # Update counter
+    increment()
+    timer.update(int(counter.value))
+    return list(urls_to_keep)
+
+
+def download_urls(deduped_urls, person):
+    """Downloads the urls to the image directory
+
+    Arguments:
+        deduped_urls {str[]} -- list of urls to download
+        person {str} -- name of the person
+    Returns:
+        None, None -- only to follow same function format
+    """
+
+    if len(deduped_urls) > 5:
+        directory = images_directory + person
+        if not os.path.exists(directory):
+            os.makedirs(directory)
```

This will have a race condition in case the directory gets created after the exists check. It's better to create the directory and ignore the exception if it already exists. Python 3.2+ even makes this very simple with the [exist_ok](https://docs.python.org/3/library/os.html#os.makedirs) flag: `makedirs(directory, exist_ok=True)`

---

fixed!